pkg/buffer/buffer.go

```diff
@@ -118,6 +123,14 @@ func NewBuffer(ssrc uint32, vp, ap *sync.Pool) *Buffer {
 func (b *Buffer) Bind(params webrtc.RTPParameters, o Options) {
 	b.Lock()
 	defer b.Unlock()
+
+	if o.Logger == nil {
+		println("Please pass logr instance implementation in Buffer Config. You can use https://github.com/pion/ion-sfu/blob/5e6f7a3f58f0411d65ef4b31e8b6d58b6afd16da/pkg/logger/zerologr.go")
+		os.Exit(1)
+	}
+	defaultLogger = o.Logger
+	debugLogger = o.Logger.V(1)
```

Comment:
I am not sure, that different log formats are good solution. It might be painful enough for the people who use sfu and want to process somehow their logs.

So, we can do the trick with public Logger instance, but ask users to reinit it with own logr if they want to have same log format everywhere

Reply:
> that different log formats are good solution

Can you elaborate on what you mean?  I don't think either suggestion changes how easy or hard it is to process logging output?  I expect users will want to set their own loggers (doable simply via `pkgfoo.Logger = MyOwnLogger()` somewhere in their binary's main pkg)?  That is certainly, for us, what is motivating the change; we have no interest in the logging implemented in the package here.

Reply:
Sorry, are NOT good solution, I mean. Never mind, we will show the user Warning and create DiscardLogger if user did not provide own one.

I made defaultLogger global in buffer package, for example, because like [here](https://github.com/pion/ion-sfu/blob/aa347bbaa0151503bbeded9d04989e8daf797785/pkg/buffer/helpers.go#L168) - isH264Keyframe - it's just a function, not a method that can use logger as a property.

What is the best way to deal with cases like that? WDYT? @bshihr

Reply:
> I made defaultLogger global in buffer package, for example, because like here - isH264Keyframe - it's just a function, not a method that can use logger as a property.

Yea that's why I suggested the second option of just upgrading it to a public package variable (`Logger`) with a silent default (DiscardLogger) and using it everywhere which would require end-users to assign their desired values.

Reply:
Done